x-pack/plugins/security_solution/server/endpoint/services/endpoint_fleet_services.ts

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { KibanaRequest } from 'kibana/server';
+import {
+  AgentClient,
+  AgentPolicyServiceInterface,
+  FleetStartContract,
+  PackagePolicyServiceInterface,
+  PackageService,
+} from '../../../../fleet/server';
+
+export interface EndpointFleetServicesFactoryInterface {
+  asScoped(req: KibanaRequest): EndpointScopedFleetServicesInterface;
+
+  asInternalUser(): EndpointInternalFleetServicesInterface;
+}
+
+export class EndpointFleetServicesFactory implements EndpointFleetServicesFactoryInterface {
+  constructor(
+    private readonly fleetDependencies: Pick<
+      FleetStartContract,
+      'agentService' | 'packageService' | 'packagePolicyService' | 'agentPolicyService'
+    >
+  ) {}
+
+  asScoped(req: KibanaRequest): EndpointScopedFleetServicesInterface {
+    const {
```

Comment:
FYI: Endpoint team: 
I would like at some point to maybe use this to also scope the the other services from fleet. Each of those normally takes in either an ES scoped client or a scoped SO client, so we could adjust the other services here so that we no longer have to pass in these client explicitly.  Example for the `packagePolicy.list()` method:

```typescript
{
  list: packagePolicy.list.bind(packagePolicy, soScopedClient)
}
```

and then we could just `packagePolicy.list(options)` - no more need to pass in the decencies (in this case the SO Client)